hbase-server/src/main/java/org/apache/hadoop/hbase/replication/BaseReplicationEndpoint.java

```diff
@@ -103,6 +103,12 @@ protected WALEntryFilter getScopeWALEntryFilter() {
   /** Returns a WALEntryFilter for checking replication per table and CF. Subclasses can
    * return null if they don't want this filter */
   protected WALEntryFilter getNamespaceTableCfWALEntryFilter() {
+    //If none of the below sets are defined, there's no reason to create this filter
+    if(ctx.getPeerConfig().getNamespaces()==null && ctx.getPeerConfig().getTableCFsMap()==null
```

Comment:
Should we handle the empty values as well?
`Strings.isNullOrEmpty(strVar)`

Reply:
Ack! Switched to `CollectionUtils.isEmpty` and `MapUtils.isEmpty`.